src/ws/cockpithandlers.c

```diff
@@ -261,21 +262,64 @@ build_environment (GHashTable *os_release)
   return g_byte_array_free_to_bytes (buffer);
 }
 
+static GBytes *
+build_oauth (CockpitAuth *auth)
+{
+  static const gchar *prefix;
+  static const gchar *suffix = ";\n    </script>";
+  static const gchar *url;
+  GByteArray *buffer = NULL;
+  GBytes *bytes = NULL;
+  JsonObject *object = NULL;
+
+  url = cockpit_conf_string ("oauth", "url");
+  if (url)
+    {
+      prefix = "\n    <script>\nvar oauth = ";
```

Comment:
There should only be one environment block. The inject hack is bad enough as it is. That is, if pages could read their own headers, I would have sent all these things as headers ... but pages can't.